src/Controls/GlobeControls.js

```diff
@@ -759,26 +770,36 @@ class GlobeControls extends THREE.EventDispatcher {
         }
     }
 
-    handleZoom(event) {
+    handleZoom(event, dollyOrigin) {
         this.player.stop();
         CameraUtils.stop(this.view, this.camera);
 
-        this.updateTarget();
-        const delta = -event.delta;
-        this.dolly(delta);
+        var point = dollyOrigin ?? this.view.getPickingPositionFromDepth(event.viewCoords);        // mouse position
```

Comment:
There's a case where the `dollyOrigin` is `null`

Reply:
could you rename `point` with explicit name, `pickedPoint`?

Reply:
If dolly, then I don't update `pickedPoint` value. So I can use a boolean instead of dollyOrigin. What do you think ?
```suggestion
        var pickedPoint = dollyFlag ?? this.view.getPickingPositionFromDepth(event.viewCoords);        // mouse position
```